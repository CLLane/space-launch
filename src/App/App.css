nav {
  height: 15vh;
}

main { 
  display: flex;
}
aside {
  display: flex;
  flex-direction: column;
  width: 15%;
}
section {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  overflow: hidden;
  width: 85%;
}

article {
  min-width: 30vw;
  min-height: 50vh;
  margin-bottom: 27px;
  text-overflow: clip;
  background-color: rgb(0, 0, 255, .5);
  border: 2px black solid;
}