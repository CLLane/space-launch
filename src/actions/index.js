export const nextLaunchData = (launch) => ({
  type: 'NEXT_LAUNCH',
  launch,
})

export const monthLaunchData = (launches) => ({
  type: 'MONTH_LAUNCH',
  launches
})

export const rocketTypeData = (rockets) => ({
  type: 'ROCKET_TYPE',
  rockets
})

export const getMissionData = (missions) => ({
  type: 'MISSION_DATA',
  missions
})

export const getCompanyData = (companies) => ({
  type: 'COMPANY_DATA',
  companies
})