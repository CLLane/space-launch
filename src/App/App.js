import React from "react";
import { Component } from "react";
import { getLaunchSchedule, monthLaunchSchedule } from "../apiCalls/apiCalls";
import ScheduleContainer from "../ScheduleContainer/ScheduleContainer";
import  LandingContainer from "../LandingContainer/LandingContainer";
import RocketTypeContainer from "../RocketTypeContainer/RocketTypeContainer";
import { Route, NavLink } from "react-router-dom";
import "./App.css";
import CompanyContainer from "../CompanyContainer/CompanyContainer";
import MissionsContainer from "../MissionsContainer/MissionsContainer";
import { connect } from "react-redux";
import { nextLaunchData } from "../actions/index";
import { monthLaunchData } from "../actions/index";
import Nav from "../Nav/nav";
import SearchContainer from "../SearchContainer/searchContainer";

export class App extends Component {

  async componentDidMount() {
    let launches = await getLaunchSchedule(1);
    this.props.nextLaunchData(launches);

    let monthStart = await this.formatMonthStart();
    let monthEnd = await this.formatMonthEnd();
    let month = await monthLaunchSchedule(monthStart, monthEnd);
    this.props.monthLaunchData(month);
  }

  formatMonthStart = () => {
    let date = new Date();
    let month = date.getMonth() + 1;
    if (month < 10) {
      month = `0${month}`;
    }
    let day = date.getDate();
    let year = date.getFullYear();
    return `${year}-${month}-${day}`;
  };

  formatMonthEnd = () => {
    let date = new Date();
    let month = date.getMonth() + 2;
    if (month < 10) {
      month = `0${month}`;
    }
    let day = date.getDate();
    let year = date.getFullYear();
    return `${year}-${month}-${day}`;
  };

  render() {
    return (
      <body>
        <nav>
          <Nav />
        </nav>
        <main>
          <aside>
            <NavLink className='nav-link' exact to="/">
              Home
            </NavLink>
            <NavLink className='nav-link' exact to="/Schedule">
              Schedule
            </NavLink>
            <NavLink className='nav-link' exact to="/RocketTypes">
              Rocket Types
            </NavLink>
            <NavLink className='nav-link' exact to="/Companies">
              Companies
            </NavLink>
            <NavLink className='nav-link' exact to="/Missions">
              Missions
            </NavLink>
          </aside>
          <section>
            {this.props.nextLaunch && (
              <Route
                exact
                path="/"
                render={() => <LandingContainer />}
              />
            )}
            {this.props.monthLaunch && (
              <Route
                exact
                path="/Schedule"
                render={() => (
                  <ScheduleContainer />
                )}
              />
            )}
            <Route
              exact
              path="/RocketTypes"
              render={() => <RocketTypeContainer />}
            />
            <Route
              exact
              path="/Companies"
              render={() => 
                <CompanyContainer />} 
            />
            <Route 
              exact
              path='/Missions'
              render={() => <MissionsContainer />}/>
            <Route 
              exact
              path='/Search'
              render={() => <SearchContainer />}/>
          </section>
        </main>
      </body>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  nextLaunchData: (nextLaunch) => dispatch(nextLaunchData(nextLaunch)),
  monthLaunchData: (monthLaunch) => dispatch(monthLaunchData(monthLaunch))
})

export const mapStateToProps = ({ nextLaunch, monthLaunch }) => ({
  nextLaunch,
  monthLaunch
})


export default connect (mapStateToProps, mapDispatchToProps) (App);
