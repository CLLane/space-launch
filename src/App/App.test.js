import React from 'react';
import {App, mapDispatchToProps, mapStateToProps} from './App';

import {shallow} from 'enzyme';
import { nextLaunchData, monthLaunchData } from "../actions/index";

describe('App', () => {
  let wrapper;
  let mockNextLaunch = [{name: 'Steve'}]
  let monthLaunch = [{name: 'james'}, {name: 'Tony'}]
  let mockNextLaunchData = jest.fn()
  let mockMonthLaunchData = jest.fn()
  beforeEach(() => {
    wrapper = shallow(
      <App 
       nextLaunch={mockNextLaunch}
       monthLaunch={monthLaunch}
       nextLaunchData={mockNextLaunchData}
       monthLaunchData={mockMonthLaunchData}
      />
    )
  });
  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });
});

describe('mapDipatchToProps', () => {
  it('calls dispatch with a nextLaunchData action', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = nextLaunchData(["mock"]);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.nextLaunchData(["mock"]);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  });
  it('calls dispatch with a monthLaunchData action', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = monthLaunchData(["mock"]);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.monthLaunchData(["mock"]);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  });
});

describe('mapStateToProps', () => {
  it('should return an array of launch objects', () => {
    const mockState = {
      nextLaunch: [{ name: "space x" }],
      monthLaunch: [{ name: "so close" }]
    };
    const expected = {
      nextLaunch: [{ name: "space x" }],
      monthLaunch: [{name: 'so close'}]
    };
    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps).toEqual(expected)
  });
});

describe('componentDidMount', () => {
  it('should invoke methods when component mounts', () => {
   
  })
})