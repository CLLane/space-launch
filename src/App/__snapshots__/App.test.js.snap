// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match a snapshot 1`] = `
<body>
  <nav>
    <Connect(Nav) />
  </nav>
  <main>
    <aside>
      <NavLink
        className="nav-link"
        exact={true}
        to="/"
      >
        Home
      </NavLink>
      <NavLink
        className="nav-link"
        exact={true}
        to="/Schedule"
      >
        Schedule
      </NavLink>
      <NavLink
        className="nav-link"
        exact={true}
        to="/RocketTypes"
      >
        Rocket Types
      </NavLink>
      <NavLink
        className="nav-link"
        exact={true}
        to="/Companies"
      >
        Companies
      </NavLink>
      <NavLink
        className="nav-link"
        exact={true}
        to="/Missions"
      >
        Missions
      </NavLink>
    </aside>
    <section>
      <Route
        exact={true}
        path="/"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/Schedule"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/RocketTypes"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/Companies"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/Missions"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/Search"
        render={[Function]}
      />
    </section>
  </main>
</body>
`;
